external/torch-mlir/lib/Dialect/Torch/Transforms/ReduceOpVariants.cpp

```diff
@@ -33,12 +33,48 @@ class ConvertToImmutableTensors : public RewritePattern {
       // Convert all operands.
       SmallVector<Value> newOperands;
       for (OpOperand &opOperand : op->getOpOperands()) {
-        auto tensorType =
-            opOperand.get().getType().dyn_cast<NonValueTensorType>();
-        if (!tensorType)
-          continue;
-        opOperand.set(rewriter.create<CopyToValueTensorOp>(op->getLoc(),
-                                                           opOperand.get()));
+        Type operandType = opOperand.get().getType();
+        if (operandType.isa<NonValueTensorType>())
+          opOperand.set(rewriter.create<CopyToValueTensorOp>(op->getLoc(),
+                                                             opOperand.get()));
+        else if (auto listType = operandType.dyn_cast<ListType>()) {
+          if (!listType.getContainedType().isa<NonValueTensorType>())
+            continue;
+
+          // Construct a new list whose elements are value tensors copied from
+          // the none value tensors of the original list.
+          auto listConstruct =
+              opOperand.get().getDefiningOp<PrimListConstructOp>();
+          if (!listConstruct || listConstruct.elements().empty())
+            continue;
```

Comment:
is it safe to continue, or do we need to return failure? A test for this case would be good.